packages/gen-typescript-declarations/src/gen-ts.ts

```diff
@@ -1037,6 +1041,29 @@ function isPolymerElement(feature: analyzer.Feature):
   return feature.kinds.has('polymer-element');
 }
 
+/**
+ * Return whether a reference looks like it is a FooBehaviorImpl style behavior
+ * object, which we want to ignore.
+ *
+ * Polymer behavior libraries are often written like:
+ *
+ *   /** @polymerBehavior FooBehavior *\/
+ *   export const FooBehaviorImpl = {};
+ *
+ *   /** @polymerBehavior *\/
+ *   export const FooBehavior = [FooBehaviorImpl, OtherBehavior];
+ *
+ * In this case, Analyzer merges FooBehaviorImpl into FooBehavior and does not
+ * emit a behavior feature for FooBehaviorImpl. However, there is still an
+ * export feature for FooBehaviorImpl, so we exclude it here.
+ */
+function isBehaviorImpl(reference: analyzer.Reference<analyzer.Feature>) {
+  return reference.feature !== undefined &&
+      reference.feature.kinds.has('behavior') &&
+      (reference.feature as analyzer.PolymerBehavior).name !==
```

Comment:
This made me wonder, could `kinds` be a `Set<keyof FeatureKindMap>` instead of `Set<string>`? means if we do this pattern often we could just have a `feature is T` function and have the correct typing. maybe? unrelated to this PR though.

Reply:
Maybe, would be nice to make these type guards easier.